app/controllers/api/vms_controller.rb

```diff
@@ -59,6 +59,17 @@ def suspend_resource(type, id = nil, _data = nil)
     end
     central_admin :suspend_resource, :suspend
 
+    def parent_choices_resource(ids)
+      parent_choices = {}
+      parent_choices[Digest::MD5.hexdigest(ids.inspect)] ||= begin
+        ems_ids = VmOrTemplate.where(:id => ids).distinct.pluck(:ems_id).compact
+        Rbac.filtered(VmOrTemplate.where(:ems_id => ems_ids).where.not(:id => ids).order(:name))
+      end
+    end
+
+    def child_choices_resource()
```

Comment:
I think this is not needed, we need just `parent_choices_resource` for https://github.com/ManageIQ/manageiq-api/pull/671

Reply:
Dude, the linked issue is called ```Get parent/children options for VMs```

Reply:
yes. it was bad link, fixed, thanks 👍

Reply:
In the UI form the selecting parent and childs is using the same options generated by this method https://github.com/ManageIQ/manageiq-ui-classic/blob/a459a4823a30b138f44a412b8c10a1f969b1b19a/app/controllers/vm_common.rb#L898 (so you can select a parent or childs from the same pool of vms)

I am not sure, if this is right, but it is what it is in UI.

Reply:
Please see what I wrote above, https://github.com/ManageIQ/manageiq-api/pull/671#issue-316007964. We're possibly changing behavior here and I'm concerned about that.

Reply:
@gtanzillo this is still a concern.

Reply:
I am not sure we are changing anything, if this is a new action. But if so, I am not able to respond that. I would be happy to remove and simplify all constraints, but from my standpoint, I just need to get the right data for the form we are refactoring.

Reply:
The API already uses different logic to do this, it's not exactly "a new action."